modules/friendlytag.js

```diff
@@ -668,227 +690,135 @@ var generateLinks = function(checkbox) {
-	'Not English': 'written in a language other than English and needs translation',
-	'One source': 'relies largely or entirely on a single source',
-	'Original research': 'contains original research',
-	'Orphan': 'linked to from no other articles',
-	'Over-coverage': 'extensive bias or disproportional coverage towards one or more specific regions',
-	'Overlinked': 'too many duplicate and/or irrelevant links to other articles',
-	'Overly detailed': 'excessive amount of intricate detail',
-	'Over-quotation': 'too many or too-lengthy quotations for an encyclopedic entry',
-	'Peacock': 'contains wording that promotes the subject in a subjective manner without adding information',
-	'POV': 'does not maintain a neutral point of view',
-	'Primary sources': 'relies too much on references to primary sources, and needs secondary sources',
-	'Prose': 'written in a list format but may read better as prose',
-	'Recentism': 'slanted towards recent events',
-	'Rough translation': 'poor translation from another language',
-	'Sections': 'needs to be divided into sections by topic',
-	'Self-published': 'contains excessive or inappropriate references to self-published sources',
-	'Sources exist': 'notable topic, sources are available that could be added to article',
-	'Technical': 'too technical for most readers to understand',
-	'Third-party': 'relies too heavily on sources too closely associated with the subject',
-	'Tone': 'tone or style may not reflect the encyclopedic tone used on Wikipedia',
-	'Too few opinions': 'may not include all significant viewpoints',
-	'Uncategorized': 'not added to any categories',
-	'Under construction': 'in the process of an expansion or major restructuring',
-	'Underlinked': 'needs more wikilinks to other articles',
-	'Undue weight': 'lends undue weight to certain ideas, incidents, or controversies',
-	'Unfocused': 'lacks focus or is about more than one topic',
-	'Unreferenced': 'does not cite any sources at all',
-	'Unreliable sources': 'some references may not be reliable',
-	'Undisclosed paid': 'may have been created or edited in return for undisclosed payments',
-	'Update': 'needs additional up-to-date information added',
-	'Very long': 'too long to read and navigate comfortably',
-	'Weasel': 'neutrality or verifiability is compromised by the use of weasel words'
-};
-
-// A list of tags in order of category
-// Tags should be in alphabetical order within the categories
+// Tags arranged by category; will be used to generate the alphabetical list,
+// but tags should be in alphabetical order within the categories
 // Add new categories with discretion - the list is long enough as is!
-
-Twinkle.tag.article.tagCategories = {
+Twinkle.tag.article.tagList = {
 	'Cleanup and maintenance tags': {
 		'General cleanup': [
-			'Cleanup',  // has a subgroup with text input
-			'Cleanup rewrite',
-			'Copy edit'  // has a subgroup with text input
+			{ tag: 'Cleanup', description: 'requires cleanup' },  // has a subgroup with text input
```

Comment:
1. It's a bit confusing that we're using `tag` & `description` here, rather than the usual `label` & `value` (for redirects/files). Have we thought about this?

2. There should be another boolean property to indicate multiple-issues exception, replicated in the `flatObject` for easy lookup. So that we can get rid of `multipleIssuesExceptions` being a separate array .

Reply:
1. No, but as you see in #1002 I went away from that for redirects; unnecessary duplication of the name, and limiting if we wanted to do something else.

2. Word!

Reply:
2: Done.  `flatObject` not so flat, could use a better name anyway